src/histolab/slide.py

```diff
@@ -76,42 +167,103 @@ def dimensions(self) -> Tuple[int, int]:
         dimensions : Tuple[int, int]
             Slide dimensions (width, height)
         """
+        if self._use_largeimage:
+            return self._metadata["sizeX"], self._metadata["sizeY"]
+
         return self._wsi.dimensions
 
     def extract_tile(
-        self, coords: CoordinatePair, level: int, tile_size: Tuple
+        self,
+        coords: CoordinatePair,
+        tile_size: Tuple[int, int],
+        level: int = None,
+        mpp: float = None,
     ) -> Tile:
         """Extract a tile of the image at the selected level.
 
         Parameters
         ----------
         coords : CoordinatePair
             Coordinates at level 0 from which to extract the tile.
+        tile_size : Tuple[int, int]
+            Final size of the extracted tile (x,y). If you choose to specify
+            the `mpp` argument, you may elect to set this as `None` to return
+            the tile as-is from `large_image` without any resizing. This is not
+            recommended, as tile size may be off by a couple of pixels when
+            coordinates are mapped to the exact mpp you request.
         level : int
-            Level from which to extract the tile.
-        tile_size: tuple
-            Final size of the tile (x,y).
+            Level from which to extract the tile. If you specify this, and
+            `mpp` is None, `openslide` will be used to fetch tiles from this
+            level from the slide. `openslide` is used for fetching tiles by
+            level, regardless of `self.use_largeimage`.
+        mpp : float
+            Micron per pixel resolution. Takes precedence over level. If this
+            is not None, `large_image` will be used to fetch tiles at the exact
+            microns-per-pixel resolution requested.
 
         Returns
         -------
         tile : Tile
             Image containing the selected tile.
         """
+        if level is None and mpp is None:
+            raise ValueError("Either level or mpp must be provided!")
+
+        if level is not None:
+            level = level if level >= 0 else self._remap_level(level)
 
-        level = level if level >= 0 else self._remap_level(level)
         if not self._has_valid_coords(coords):
             # OpenSlide doesn't complain if the coordinates for extraction are wrong,
             # but it returns an odd image.
-            raise ValueError(
+            raise TileSizeOrCoordinatesError(
                 f"Extraction Coordinates {coords} not valid for slide with dimensions "
                 f"{self.dimensions}"
             )
 
-        image = self._wsi.read_region(
-            location=(coords.x_ul, coords.y_ul), level=level, size=tile_size
-        )
-        tile = Tile(image, coords, level)
-        return tile
+        if mpp is None:
+            image = self._wsi.read_region(
```

Comment:
This will output RGBA images, while if we extract with large_image it will be converted to RGB (by hand, line 241), we should uniform this... I personally would keep it as it was originally, so RGBA

Reply:
are you suggesting `image = image.convert("RGBA")` ?

Reply:
I'm suggesting to remove line 241, if that works